straxen/plugins/acqmon_processing.py

```diff
@@ -56,163 +137,307 @@ def compute(self, raw_records_aqmon):
+        elif len(extra_stop):
+            message += ' Despite the fact that we inserted one extra stop at the end of the run.'  # noqa
+        if something_is_wrong:
+            raise ValueError(message)
+
+        if np.any(veto_hits_start['time'] > veto_hits_stop['time']):
+            raise ValueError('Found veto\'s starting before the previous stopped')
+
+        return veto_hits_start, veto_hits_stop
+
+    @staticmethod
+    def fake_hit(start, dt=1, length=1):
+        hit = np.zeros(1, strax.hit_dtype)
+        hit['time'] = start
+        hit['dt'] = dt
+        hit['length'] = length
+        return hit
+
+
+# Don't use @numba since numba doesn't like masking arrays, use numpy
+def channel_select(rr, ch):
     """Return data from start/stop veto channel in the acquisition monitor (AM)"""
     return rr[rr['channel'] == ch]
 
 
 @export
-@strax.takes_config(
-    strax.Option('veto_proximity_window', default=int(5e8), type=int, track=True,
-                 help='Maximum separation between veto stop and start pulses [ns]'))
 class VetoProximity(strax.OverlapWindowPlugin):
     """
-    Find the closest next/previous veto start and end to each event center.
-
-    previous_x: Time in ns between the time center of an event and the previous x
-    next_x: Time in ns between the time center of an event and the next x
-    This also considers any x inside the event. x could be either:
-        - busy_x: busy on/off signal
-        - he_x:   high energy channels busy on/off signal
-        - hev_x:  high energy veto on/off signal
+    Find the closest next/previous veto start w.r.t. the event time or
+    when a busy happens during an event.
     """
 
-    __version__ = '0.1.2'
-    depends_on = ('event_basics', 'aqmon_hits')
-    provides = ('veto_proximity')
-    data_kind = ('events')
-    save_when = strax.SaveWhen.TARGET
-    veto_names = ['busy', 'he', 'hev']
+    __version__ = '2.1.0'
+    # Strictly speaking, we could depend on 'events', but then you couldn't
+    # change the event_window_fields to e.g. s1_time and s2_endtime.
+    depends_on = ('event_basics', 'veto_intervals')
+    provides = 'veto_proximity'
+    data_kind = 'events'
+    save_when = strax.SaveWhen.NEVER
+
+    event_window_fields = straxen.URLConfig(
+        default=('time', 'endtime'),
+        help='Fields to determine where to look for overlaps for using '
+             'this plugin in the events. The default uses start and endtime '
+             'of an event, but this can also be the S1 or S2 start/endtime'
+    )
+
+    veto_proximity_window = straxen.URLConfig(
+        default=int(300e9),
+        help='Maximum separation between veto stop and start pulses [ns]'
+    )
+    time_no_aqmon_veto_found = straxen.URLConfig(
+        default=int(3.6e+12),
+        track=True,
+        type=int,
+        help='If we don\'t find a veto close to the event, say that '
+             'the closest event is this many ns removed from it.'
```

Comment:
Sorry, but I do not get that sentence :D

Reply:
ok I'll change it